BoxHub/config_sample_php_parameters.rst
========================
Configuration Parameters
========================

Nextcloud uses the ``config/config.php`` file to control server operations.
``config/config.sample.php`` lists all the configurable parameters within
Nextcloud, along with example or default values. This document provides a more
detailed reference. Most options are configurable on your Admin page, so it
is usually not necessary to edit ``config/config.php``.

.. note:: The installer creates a configuration containing the essential parameters.
   Only manually add configuration parameters to ``config/config.php`` if you need to
   use a special value for a parameter. **Do not copy everything from**
   ``config/config.sample.php`` **. Only enter the parameters you wish to modify!**

Multiple config.php file
------------------------

Nextcloud supports loading configuration parameters from multiple files.
You can add arbitrary files ending with :file:`.config.php` in the :file:`config/`
directory, for example you could place your email server configuration
in :file:`email.config.php`. This allows you to easily create and manage
custom configurations, or to divide a large complex configuration file
into a set of smaller files. These custom files are not overwritten by
Nextcloud, and the values in these files take precedence over :file:`config.php`.

.. The following section is auto-generated from
.. https://github.com/nextcloud/server/blob/master/config/config.sample.php
.. Do not edit this file; edit the source file in core
.. DEFAULT_SECTION_START


Default Parameters
------------------

These parameters are configured by the Nextcloud installer, and are required
for your Nextcloud server to operate.


::

	'instanceid' => '',

This is a unique identifier for your Nextcloud installation, created
automatically by the installer. This example is for documentation only,
and you should never use it because it will not work. A valid ``instanceid``
is created when you install Nextcloud.

'instanceid' => 'd3c944a9a',

::

	'passwordsalt' => '',

The salt used to hash all passwords, auto-generated by the Nextcloud
installer. (There are also per-user salts.) If you lose this salt you lose
all your passwords. This example is for documentation only, and you should
never use it.

::

	'trusted_domains' =>
	   [
	    'demo.example.org',
	    'otherdomain.example.org',
	    '10.111.112.113',
	    '[2001:db8::1]'
	  ],

Your list of trusted domains that users can log into. Specifying trusted
domains prevents host header poisoning. Do not remove this, as it performs
necessary security checks.

You can specify:

- the exact hostname of your host or virtual host, e.g. demo.example.org.
- the exact hostname with permitted port, e.g. demo.example.org:443.
  This disallows all other ports on this host
- use * as a wildcard, e.g. ubos-raspberry-pi*.local will allow
  ubos-raspberry-pi.local and ubos-raspberry-pi-2.local
- the IP address with or without permitted port, e.g. [2001:db8::1]:8080
  Using TLS certificates where commonName=<IP address> is deprecated

::

	'datadirectory' => '/var/www/nextcloud/data',

Where user files are stored. The SQLite database is also stored here, when
you use SQLite.

Default to ``data/`` in the Nextcloud directory.

::

	'version' => '',

The current version number of your Nextcloud installation. This is set up
during installation and update, so you shouldn't need to change it.

::

	'dbtype' => 'sqlite3',

Identifies the database used with this installation. See also config option
``supportedDatabases``

Available:
	- sqlite3 (SQLite3)
	- mysql (MySQL/MariaDB)
	- pgsql (PostgreSQL)

Defaults to ``sqlite3``

::

	'dbhost' => '',

Your host server name, for example ``localhost``, ``hostname``,
``hostname.example.com``, or the IP address. To specify a port use
``hostname:####``; to specify a Unix socket use
``localhost:/path/to/socket``.

::

	'dbname' => 'nextcloud',

The name of the Nextcloud database, which is set during installation. You
should not need to change this.

::

	'dbuser' => '',

The user that Nextcloud uses to write to the database. This must be unique
across Nextcloud instances using the same SQL database. This is set up during
installation, so you shouldn't need to change it.

::

	'dbpassword' => '',

The password for the database user. This is set up during installation, so
you shouldn't need to change it.

::

	'dbtableprefix' => '',

Prefix for the Nextcloud tables in the database.

Default to ``oc_``

::

	'installed' => false,

Indicates whether the Nextcloud instance was installed successfully; ``true``
indicates a successful installation, and ``false`` indicates an unsuccessful
installation.

Defaults to ``false``

.. DEFAULT_SECTION_END
.. Generated content above. Don't change this.

Default config.php Examples
---------------------------
When you use SQLite as your Nextcloud database, your ``config.php`` looks like
this after installation. The SQLite database is stored in your Nextcloud
``data/`` directory. SQLite is a simple, lightweight embedded database that
is good for testing and for simple installations, but for production Nextcloud
systems you should use MySQL, MariaDB, or PosgreSQL.

::

  <?php
  $CONFIG = array (
    'instanceid' => 'occ6f7365735',
    'passwordsalt' => '2c5778476346786306303',
    'trusted_domains' =>
    array (
      0 => 'localhost',
      1 => 'studio',
    ),
    'datadirectory' => '/var/www/nextcloud/data',
    'dbtype' => 'sqlite3',
    'version' => '7.0.2.1',
    'installed' => true,
  );

This example is from a new Nextcloud installation using MariaDB::


  <?php
  $CONFIG = array (
    'instanceid' => 'oc8c0fd71e03',
    'passwordsalt' => '515a13302a6b3950a9d0fdb970191a',
    'trusted_domains' =>
    array (
      0 => 'localhost',
      1 => 'studio',
      2 => '192.168.10.155'
    ),
    'datadirectory' => '/var/www/nextcloud/data',
    'dbtype' => 'mysql',
     'version' => '7.0.2.1',
    'dbname' => 'nextcloud',
    'dbhost' => 'localhost',
    'dbtableprefix' => 'oc_',
    'dbuser' => 'oc_carla',
    'dbpassword' => '67336bcdf7630dd80b2b81a413d07',
    'installed' => true,
  );

.. Generated content below. Don't change this.
.. ALL_OTHER_SECTIONS_START


User Experience
---------------

These optional parameters control some aspects of the user interface. Default
values, where present, are shown.


::

	'default_language' => 'en',

This sets the default language on your Nextcloud server, using ISO_639-1
language codes such as ``en`` for English, ``de`` for German, and ``fr`` for
French. It overrides automatic language detection on public pages like login
or shared items. User's language preferences configured under "personal ->
language" override this setting after they have logged in. Nextcloud has two
distinguished language codes for German, 'de' and 'de_DE'. 'de' is used for
informal German and 'de_DE' for formal German. By setting this value to 'de_DE'
you can enforce the formal version of German unless the user has chosen
something different explicitly.

Defaults to ``en``

::

	'force_language' => 'en',

With this setting a language can be forced for all users. If a language is
forced, the users are also unable to change their language in the personal
settings. If users shall be unable to change their language, but users have
different languages, this value can be set to ``true`` instead of a language
code.

Defaults to ``false``

::

	'default_locale' => 'en_US',

This sets the default locale on your Nextcloud server, using ISO_639
language codes such as ``en`` for English, ``de`` for German, and ``fr`` for
French, and ISO-3166 country codes such as ``GB``, ``US``, ``CA``, as defined
in RFC 5646. It overrides automatic locale detection on public pages like
login or shared items. User's locale preferences configured under "personal
-> locale" override this setting after they have logged in.

Defaults to ``en``

::

	'default_phone_region' => 'GB',

This sets the default region for phone numbers on your Nextcloud server,
using ISO 3166-1 country codes such as ``DE`` for Germany, ``FR`` for France, …
It is required to allow inserting phone numbers in the user profiles starting
without the country code (e.g. +49 for Germany).

No default value!

::

	'force_locale' => 'en_US',

With this setting a locale can be forced for all users. If a locale is
forced, the users are also unable to change their locale in the personal
settings. If users shall be unable to change their locale, but users have
different languages, this value can be set to ``true`` instead of a locale
code.

Defaults to ``false``

::

	'defaultapp' => 'dashboard,files',

Set the default app to open on login. Use the app names as they appear in the
URL after clicking them in the Apps menu, such as documents, calendar, and
gallery. You can use a comma-separated list of app names, so if the first
app is not enabled for a user then Nextcloud will try the second one, and so
on. If no enabled apps are found it defaults to the dashboard app.

Defaults to ``dashboard,files``

::

	'knowledgebaseenabled' => true,

``true`` enables the Help menu item in the user menu (top right of the
Nextcloud Web interface). ``false`` removes the Help item.

::

	'allow_user_to_change_display_name' => true,

``true`` allows users to change their display names (on their Personal
pages), and ``false`` prevents them from changing their display names.

::

	'remember_login_cookie_lifetime' => 60*60*24*15,

Lifetime of the remember login cookie. This should be larger than the
session_lifetime. If it is set to 0 remember me is disabled.

Defaults to ``60*60*24*15`` seconds (15 days)

::

	'session_lifetime' => 60 * 60 * 24,

The lifetime of a session after inactivity.

Defaults to ``60*60*24`` seconds (24 hours)

::

	'session_keepalive' => true,

Enable or disable session keep-alive when a user is logged in to the Web UI.

Enabling this sends a "heartbeat" to the server to keep it from timing out.

Defaults to ``true``

::

	'auto_logout' => false,

Enable or disable the automatic logout after session_lifetime, even if session
keepalive is enabled. This will make sure that an inactive browser will be logged out
even if requests to the server might extend the session lifetime.

Defaults to ``false``

::

	'token_auth_enforced' => false,

Enforce token authentication for clients, which blocks requests using the user
password for enhanced security. Users need to generate tokens in personal settings
which can be used as passwords on their clients.

Defaults to ``false``

::

	'token_auth_activity_update' => 60,

The interval at which token activity should be updated.

Increasing this value means that the last activty on the security page gets
more outdated.

Tokens are still checked every 5 minutes for validity
max value: 300

Defaults to ``300``

::

	'auth.bruteforce.protection.enabled' => true,

Whether the bruteforce protection shipped with Nextcloud should be enabled or not.

Disabling this is discouraged for security reasons.

Defaults to ``true``

::

	'auth.webauthn.enabled' => true,

By default WebAuthn is available but it can be explicitly disabled by admins

::

	'skeletondirectory' => '/path/to/nextcloud/core/skeleton',

The directory where the skeleton files are located. These files will be
copied to the data directory of new users. Leave empty to not copy any
skeleton files.

``{lang}`` can be used as a placeholder for the language of the user.
If the directory does not exist, it falls back to non dialect (from ``de_DE``
to ``de``). If that does not exist either, it falls back to ``default``

Defaults to ``core/skeleton`` in the Nextcloud directory.

::

	'templatesdirectory' => '/path/to/nextcloud/templates',

The directory where the template files are located. These files will be
copied to the template directory of new users. Leave empty to not copy any
template files.

``{lang}`` can be used as a placeholder for the language of the user.
If the directory does not exist, it falls back to non dialect (from ``de_DE``
to ``de``). If that does not exist either, it falls back to ``default``

If this is not set creating a template directory will only happen if no custom
``skeletondirectory`` is defined, otherwise the shipped templates will be used
to create a template directory for the user.

::

	'lost_password_link' => 'https://example.org/link/to/password/reset',

If your user backend does not allow password resets (e.g. when it's a
read-only user backend like LDAP), you can specify a custom link, where the
user is redirected to, when clicking the "reset password" link after a failed
login-attempt.

In case you do not want to provide any link, replace the url with 'disabled'

Mail Parameters
---------------

These configure the email settings for Nextcloud notifications and password
resets.


::

	'mail_domain' => 'example.com',

The return address that you want to appear on emails sent by the Nextcloud
server, for example ``nc-admin@example.com``, substituting your own domain,
of course.

::

	'mail_from_address' => 'nextcloud',

FROM address that overrides the built-in ``sharing-noreply`` and
``lostpassword-noreply`` FROM addresses.

Defaults to different from addresses depending on the feature.

::

	'mail_smtpdebug' => false,

Enable SMTP class debugging.

Defaults to ``false``

::

	'mail_smtpmode' => 'smtp',

Which mode to use for sending mail: ``sendmail``, ``smtp`` or ``qmail``.

If you are using local or remote SMTP, set this to ``smtp``.

For the ``sendmail`` option you need an installed and working email system on
the server, with ``/usr/sbin/sendmail`` installed on your Unix system.

For ``qmail`` the binary is /var/qmail/bin/sendmail, and it must be installed
on your Unix system.

Defaults to ``smtp``

::

	'mail_smtphost' => '127.0.0.1',

This depends on ``mail_smtpmode``. Specify the IP address of your mail
server host. This may contain multiple hosts separated by a semi-colon. If
you need to specify the port number append it to the IP address separated by
a colon, like this: ``127.0.0.1:24``.

Defaults to ``127.0.0.1``

::

	'mail_smtpport' => 25,

This depends on ``mail_smtpmode``. Specify the port for sending mail.

Defaults to ``25``

::

	'mail_smtptimeout' => 10,

This depends on ``mail_smtpmode``. This sets the SMTP server timeout, in
seconds. You may need to increase this if you are running an anti-malware or
spam scanner.

Defaults to ``10`` seconds

::

	'mail_smtpsecure' => '',

This depends on ``mail_smtpmode``. Specify when you are using ``ssl`` for SSL/TLS or
``tls`` for STARTTLS, or leave empty for no encryption.

Defaults to ``''`` (empty string)

::

	'mail_smtpauth' => false,

This depends on ``mail_smtpmode``. Change this to ``true`` if your mail
server requires authentication.

Defaults to ``false``

::

	'mail_smtpauthtype' => 'LOGIN',

This depends on ``mail_smtpmode``. If SMTP authentication is required, choose
the authentication type as ``LOGIN`` or ``PLAIN``.

Defaults to ``LOGIN``

::

	'mail_smtpname' => '',

This depends on ``mail_smtpauth``. Specify the username for authenticating to
the SMTP server.

Defaults to ``''`` (empty string)

::

	'mail_smtppassword' => '',

This depends on ``mail_smtpauth``. Specify the password for authenticating to
the SMTP server.

Default to ``''`` (empty string)

::

	'mail_template_class' => '\OC\Mail\EMailTemplate',

Replaces the default mail template layout. This can be utilized if the
options to modify the mail texts with the theming app is not enough.

The class must extend  ``\OC\Mail\EMailTemplate``

::

	'mail_send_plaintext_only' => false,

Email will be send by default with an HTML and a plain text body. This option
allows to only send plain text emails.

::

	'mail_smtpstreamoptions' => [],

This depends on ``mail_smtpmode``. Array of additional streams options that
will be passed to underlying Swift mailer implementation.

Defaults to an empty array.

::

	'mail_sendmailmode' => 'smtp',

Which mode is used for sendmail/qmail: ``smtp`` or ``pipe``.

For ``smtp`` the sendmail binary is started with the parameter ``-bs``:
  - Use the SMTP protocol on standard input and output.

For ``pipe`` the binary is started with the parameters ``-t``:
  - Read message from STDIN and extract recipients.

Defaults to ``smtp``

Proxy Configurations
--------------------


::

	'overwritehost' => '',

The automatic hostname detection of Nextcloud can fail in certain reverse
proxy and CLI/cron situations. This option allows you to manually override
the automatic detection; for example ``www.example.com``, or specify the port
``www.example.com:8080``.

::

	'overwriteprotocol' => '',

When generating URLs, Nextcloud attempts to detect whether the server is
accessed via ``https`` or ``http``. However, if Nextcloud is behind a proxy
and the proxy handles the ``https`` calls, Nextcloud would not know that
``ssl`` is in use, which would result in incorrect URLs being generated.

Valid values are ``http`` and ``https``.

::

	'overwritewebroot' => '',

Nextcloud attempts to detect the webroot for generating URLs automatically.

For example, if ``www.example.com/nextcloud`` is the URL pointing to the
Nextcloud instance, the webroot is ``/nextcloud``. When proxies are in use,
it may be difficult for Nextcloud to detect this parameter, resulting in
invalid URLs.

::

	'overwritecondaddr' => '',

This option allows you to define a manual override condition as a regular
expression for the remote IP address. For example, defining a range of IP
addresses starting with ``10.0.0.`` and ending with 1 to 3:
``^10\.0\.0\.[1-3]$``

Defaults to ``''`` (empty string)

::

	'overwrite.cli.url' => '',

Use this configuration parameter to specify the base URL for any URLs which
are generated within Nextcloud using any kind of command line tools (cron or
occ). The value should contain the full base URL:
``https://www.example.com/nextcloud``

Defaults to ``''`` (empty string)

::

	'htaccess.RewriteBase' => '/',

To have clean URLs without `/index.php` this parameter needs to be configured.

This parameter will be written as "RewriteBase" on update and installation of
Nextcloud to your `.htaccess` file. While this value is often simply the URL
path of the Nextcloud installation it cannot be set automatically properly in
every scenario and needs thus some manual configuration.

In a standard Apache setup this usually equals the folder that Nextcloud is
accessible at. So if Nextcloud is accessible via "https://mycloud.org/nextcloud"
the correct value would most likely be "/nextcloud". If Nextcloud is running
under "https://mycloud.org/" then it would be "/".

Note that the above rule is not valid in every case, as there are some rare setup
cases where this may not apply. However, to avoid any update problems this
configuration value is explicitly opt-in.

After setting this value run `occ maintenance:update:htaccess`. Now, when the
following conditions are met Nextcloud URLs won't contain `index.php`:

- `mod_rewrite` is installed
- `mod_env` is installed

Defaults to ``''`` (empty string)

::

	'htaccess.IgnoreFrontController' => false,

For server setups, that don't have `mod_env` enabled or restricted (e.g. suEXEC)
this parameter has to be set to true and will assume mod_rewrite.

Please check, if `mod_rewrite` is active and functional before setting this
parameter and you updated your .htaccess with `occ maintenance:update:htaccess`.
Otherwise your nextcloud installation might not be reachable anymore.
For example, try accessing resources by leaving out `index.php` in the URL.

::

	'proxy' => '',

The URL of your proxy server, for example ``proxy.example.com:8081``.

Note: Guzzle (the http library used by Nextcloud) is reading the environment
variables HTTP_PROXY (only for cli request), HTTPS_PROXY, and NO_PROXY by default.

If you configure proxy with Nextcloud any default configuration by Guzzle
is overwritten. Make sure to set ``proxyexclude`` accordingly if necessary.

Defaults to ``''`` (empty string)

::

	'proxyuserpwd' => '',

The optional authentication for the proxy to use to connect to the internet.

The format is: ``username:password``.

Defaults to ``''`` (empty string)

::

	'proxyexclude' => [],

List of host names that should not be proxied to.

For example: ``['.mit.edu', 'foo.com']``.

Hint: Use something like ``explode(',', getenv('NO_PROXY'))`` to sync this
value with the global NO_PROXY option.

Defaults to empty array.

::

	'allow_local_remote_servers' => true,

Allow remote servers with local addresses e.g. in federated shares, webcal services and more

Defaults to false

Deleted Items (trash bin)
-------------------------

These parameters control the Deleted files app.


::

	'trashbin_retention_obligation' => 'auto',

If the trash bin app is enabled (default), this setting defines the policy
for when files and folders in the trash bin will be permanently deleted.

The app allows for two settings, a minimum time for trash bin retention,
and a maximum time for trash bin retention.
Minimum time is the number of days a file will be kept, after which it
may be deleted. Maximum time is the number of days at which it is guaranteed
to be deleted.
Both minimum and maximum times can be set together to explicitly define
file and folder deletion. For migration purposes, this setting is installed
initially set to "auto", which is equivalent to the default setting in
Nextcloud.

Available values:

* ``auto``
    default setting. keeps files and folders in the trash bin for 30 days
    and automatically deletes anytime after that if space is needed (note:
    files may not be deleted if space is not needed).
* ``D, auto``
    keeps files and folders in the trash bin for D+ days, delete anytime if
    space needed (note: files may not be deleted if space is not needed)
* ``auto, D``
    delete all files in the trash bin that are older than D days
    automatically, delete other files anytime if space needed
* ``D1, D2``
    keep files and folders in the trash bin for at least D1 days and
    delete when exceeds D2 days (note: files will not be deleted automatically if space is needed)
* ``disabled``
    trash bin auto clean disabled, files and folders will be kept forever

Defaults to ``auto``

File versions
-------------

These parameters control the Versions app.


::

	'versions_retention_obligation' => 'auto',

If the versions app is enabled (default), this setting defines the policy
for when versions will be permanently deleted.

The app allows for two settings, a minimum time for version retention,
and a maximum time for version retention.
Minimum time is the number of days a version will be kept, after which it
may be deleted. Maximum time is the number of days at which it is guaranteed
to be deleted.
Both minimum and maximum times can be set together to explicitly define
version deletion. For migration purposes, this setting is installed
initially set to "auto", which is equivalent to the default setting in
Nextcloud.

Available values:

* ``auto``
    default setting. Automatically expire versions according to expire
    rules. Please refer to :doc:`../configuration_files/file_versioning` for
    more information.
* ``D, auto``
    keep versions at least for D days, apply expire rules to all versions
    that are older than D days
* ``auto, D``
    delete all versions that are older than D days automatically, delete
    other versions according to expire rules
* ``D1, D2``
    keep versions for at least D1 days and delete when exceeds D2 days
* ``disabled``
    versions auto clean disabled, versions will be kept forever

Defaults to ``auto``

Nextcloud Verifications
-----------------------

Nextcloud performs several verification checks. There are two options,
``true`` and ``false``.


::

	'appcodechecker' => true,

Checks an app before install whether it uses private APIs instead of the
proper public APIs. If this is set to true it will only allow to install or
enable apps that pass this check.

Defaults to ``false``

::

	'updatechecker' => true,

Check if Nextcloud is up-to-date and shows a notification if a new version is
available.

Defaults to ``true``

::

	'updater.server.url' => 'https://updates.nextcloud.com/updater_server/',

URL that Nextcloud should use to look for updates

Defaults to ``https://updates.nextcloud.com/updater_server/``

::

	'updater.release.channel' => 'stable',

The channel that Nextcloud should use to look for updates

Supported values:
  - ``daily``
  - ``beta``
  - ``stable``

::

	'has_internet_connection' => true,

Is Nextcloud connected to the Internet or running in a closed network?

Defaults to ``true``

::

	'connectivity_check_domains' => [
		'www.nextcloud.com',
		'www.startpage.com',
		'www.eff.org',
		'www.edri.org'
	],

Which domains to request to determine the availability of an Internet
connection. If none of these hosts are reachable, the administration panel
will show a warning. Set to an empty list to not do any such checks (warning
will still be shown).

Defaults to the following domains:

 - www.nextcloud.com
 - www.startpage.com
 - www.eff.org
 - www.edri.org

::

	'check_for_working_wellknown_setup' => true,

Allows Nextcloud to verify a working .well-known URL redirects. This is done
by attempting to make a request from JS to
https://your-domain.com/.well-known/caldav/

Defaults to ``true``

::

	'check_for_working_htaccess' => true,

This is a crucial security check on Apache servers that should always be set
to ``true``. This verifies that the ``.htaccess`` file is writable and works.

If it is not, then any options controlled by ``.htaccess``, such as large
file uploads, will not work. It also runs checks on the ``data/`` directory,
which verifies that it can't be accessed directly through the Web server.

Defaults to ``true``

::

	'check_data_directory_permissions' => true,

In rare setups (e.g. on Openshift or docker on windows) the permissions check
might block the installation while the underlying system offers no means to
"correct" the permissions. In this case, set the value to false.

In regular cases, if issues with permissions are encountered they should be
adjusted accordingly. Changing the flag is discouraged.

Defaults to ``true``

::

	'config_is_read_only' => false,

In certain environments it is desired to have a read-only configuration file.

When this switch is set to ``true`` Nextcloud will not verify whether the
configuration is writable. However, it will not be possible to configure
all options via the Web interface. Furthermore, when updating Nextcloud
it is required to make the configuration file writable again for the update
process.

Defaults to ``false``

Logging
-------


::

	'log_type' => 'file',

This parameter determines where the Nextcloud logs are sent.

``file``: the logs are written to file ``nextcloud.log`` in the default
Nextcloud data directory. The log file can be changed with parameter
``logfile``.
``syslog``: the logs are sent to the system log. This requires a syslog daemon
to be active.
``errorlog``: the logs are sent to the PHP ``error_log`` function.
``systemd``: the logs are sent to the Systemd journal. This requires a system
that runs Systemd and the Systemd journal. The PHP extension ``systemd``
must be installed and active.

Defaults to ``file``

::

	'logfile' => '/var/log/nextcloud.log',

Name of the file to which the Nextcloud logs are written if parameter
``log_type`` is set to ``file``.

Defaults to ``[datadirectory]/nextcloud.log``

::

	'logfilemode' => 0640,

Log file mode for the Nextcloud loggin type in octal notation.

Defaults to 0640 (writeable by user, readable by group).

::

	'loglevel' => 2,

Loglevel to start logging at. Valid values are: 0 = Debug, 1 = Info, 2 =
Warning, 3 = Error, and 4 = Fatal. The default value is Warning.

Defaults to ``2``

::

	'syslog_tag' => 'Nextcloud',

If you maintain different instances and aggregate the logs, you may want
to distinguish between them. ``syslog_tag`` can be set per instance
with a unique id. Only available if ``log_type`` is set to ``syslog`` or
``systemd``.

The default value is ``Nextcloud``.

::

	'log.condition' => [
		'shared_secret' => '57b58edb6637fe3059b3595cf9c41b9',
		'users' => ['sample-user'],
		'apps' => ['files'],
	],

Log condition for log level increase based on conditions. Once one of these
conditions is met, the required log level is set to debug. This allows to
debug specific requests, users or apps

Supported conditions:
 - ``shared_secret``: if a request parameter with the name `log_secret` is set to
               this value the condition is met
 - ``users``:  if the current request is done by one of the specified users,
               this condition is met
 - ``apps``:   if the log message is invoked by one of the specified apps,
               this condition is met

Defaults to an empty array.

::

	'logdateformat' => 'F d, Y H:i:s',

This uses PHP.date formatting; see https://www.php.net/manual/en/function.date.php

Defaults to ISO 8601 ``2005-08-15T15:52:01+00:00`` - see \DateTime::ATOM
(https://www.php.net/manual/en/class.datetime.php#datetime.constants.atom)

::

	'logtimezone' => 'Europe/Berlin',

The timezone for logfiles. You may change this; see
https://www.php.net/manual/en/timezones.php

Defaults to ``UTC``

::

	'log_query' => false,

Append all database queries and parameters to the log file. Use this only for
debugging, as your logfile will become huge.

::

	'log_rotate_size' => 100 * 1024 * 1024,

Enables log rotation and limits the total size of logfiles. Set it to 0 for
no rotation. Specify a size in bytes, for example 104857600 (100 megabytes
= 100 * 1024 * 1024 bytes). A new logfile is created with a new name when the
old logfile reaches your limit. If a rotated log file is already present, it
will be overwritten.

Defaults to 100 MB

Alternate Code Locations
------------------------

Some of the Nextcloud code may be stored in alternate locations.


::

	'customclient_desktop' =>
		'https://nextcloud.com/install/#install-clients',
	'customclient_android' =>
		'https://play.google.com/store/apps/details?id=com.nextcloud.client',
	'customclient_ios' =>
		'https://itunes.apple.com/us/app/nextcloud/id1125420102?mt=8',
	'customclient_ios_appid' =>
			'1125420102',

This section is for configuring the download links for Nextcloud clients, as
seen in the first-run wizard and on Personal pages.

Defaults to:
 - Desktop client: ``https://nextcloud.com/install/#install-clients``
 - Android client: ``https://play.google.com/store/apps/details?id=com.nextcloud.client``
 - iOS client: ``https://itunes.apple.com/us/app/nextcloud/id1125420102?mt=8``
 - iOS client app id: ``1125420102``

Apps
----

Options for the Apps folder, Apps store, and App code checker.


::

	'appstoreenabled' => true,

When enabled, admins may install apps from the Nextcloud app store.

Defaults to ``true``

::

	'appstoreurl' => 'https://apps.nextcloud.com/api/v1',

Enables the installation of apps from a self hosted apps store.

Requires that at least one of the configured apps directories is writeable.

Defaults to ``https://apps.nextcloud.com/api/v1``

::

	'apps_paths' => [
		[
			'path'=> '/var/www/nextcloud/apps',
			'url' => '/apps',
			'writable' => true,
		],
	],

Use the ``apps_paths`` parameter to set the location of the Apps directory,
which should be scanned for available apps, and where user-specific apps
should be installed from the Apps store. The ``path`` defines the absolute
file system path to the app folder. The key ``url`` defines the HTTP Web path
to that folder, starting from the Nextcloud webroot. The key ``writable``
indicates if a Web server can write files to that folder.

::

	'appcodechecker' => true,

Checks an app before install whether it uses private APIs instead of the
proper public APIs. If this is set to true it will only allow to install or
enable apps that pass this check.

Defaults to ``false``





Previews
--------

Nextcloud supports previews of image files, the covers of MP3 files, and text
files. These options control enabling and disabling previews, and thumbnail
size.


::

	'enable_previews' => true,

By default, Nextcloud can generate previews for the following filetypes:

- Image files
- Covers of MP3 files
- Text documents

Valid values are ``true``, to enable previews, or
``false``, to disable previews

Defaults to ``true``

::

	'preview_max_x' => 4096,

The maximum width, in pixels, of a preview. A value of ``null`` means there
is no limit.

Defaults to ``4096``

::

	'preview_max_y' => 4096,

The maximum height, in pixels, of a preview. A value of ``null`` means there
is no limit.

Defaults to ``4096``

::

	'preview_max_filesize_image' => 50,

max file size for generating image previews with imagegd (default behavior)
If the image is bigger, it'll try other preview generators, but will most
likely show the default mimetype icon. Set to -1 for no limit.

Defaults to ``50`` megabytes

::

	'preview_libreoffice_path' => '/usr/bin/libreoffice',

custom path for LibreOffice/OpenOffice binary

Defaults to ``''`` (empty string)

::

	'preview_office_cl_parameters' =>
		' --headless --nologo --nofirststartwizard --invisible --norestore '.
		'--convert-to png --outdir ',

Use this if LibreOffice/OpenOffice requires additional arguments.

Defaults to ``''`` (empty string)

::

	'enabledPreviewProviders' => [
		'OC\Preview\PNG',
		'OC\Preview\JPEG',
		'OC\Preview\GIF',
		'OC\Preview\HEIC',
		'OC\Preview\BMP',
		'OC\Preview\XBitmap',
		'OC\Preview\MP3',
		'OC\Preview\TXT',
		'OC\Preview\MarkDown',
		'OC\Preview\OpenDocument',
		'OC\Preview\Krita',
	],

Only register providers that have been explicitly enabled

The following providers are disabled by default due to performance or privacy
concerns:

 - OC\\Preview\\Illustrator
 - OC\\Preview\\Movie
 - OC\\Preview\\MSOffice2003
 - OC\\Preview\\MSOffice2007
 - OC\\Preview\\MSOfficeDoc
 - OC\\Preview\\PDF
 - OC\\Preview\\Photoshop
 - OC\\Preview\\Postscript
 - OC\\Preview\\StarOffice
 - OC\\Preview\\SVG
 - OC\\Preview\\TIFF
 - OC\\Preview\\Font


Defaults to the following providers:

 - OC\\Preview\\BMP
 - OC\\Preview\\GIF
 - OC\\Preview\\HEIC
 - OC\\Preview\\JPEG
 - OC\\Preview\\MarkDown
 - OC\\Preview\\MP3
 - OC\\Preview\\PNG
 - OC\\Preview\\TXT
 - OC\\Preview\\XBitmap
 - OC\\Preview\\OpenDocument
 - OC\\Preview\\Krita

LDAP
----


BoxHub/config.php
<?php
$CONFIG = array (
  'htaccess.RewriteBase' => '/boxhub',
  'memcache.local' => '\\OC\\Memcache\\APCu',
  'apps_paths' => 
  array (
    0 => 
    array (
      'path' => '/var/www/html/boxapps',
      'url' => '/apps',
      'writable' => false,
    ),
    1 => 
    array (
      'path' => '/var/www/html/custom_apps',
      'url' => '/custom_apps',
      'writable' => false,
    ),
  ),
  'installed' => false,
  'trusted_domains' => 
  array (
    0 => 'localhost:8082',
    1 => '*.homehub_frontend',
    2 => 'localhost',
  ),
  'datadirectory' => '/var/www/html/files',
  'dbtype' => 'mysql',
  'version' => '20.0.5.2',
  'overwrite.cli.url' => '',
  'dbname' => '_hbox_',
  'dbhost' => 'homehub_sql_1',
  'dbport' => '3306',
  'dbtableprefix' => '[hb]:',
  'mysql.utf8mb4' => true,
  'dbuser' => 'u[box]',
  'dbpassword' => 'u{zO[6HqVl7I[-y{t6stiF05KFW1H[FKkZETg-ef',
  'installed' => false,
  'default_language' => 'it',
  'allow_user_to_change_display_name' => true,
  'remember_login_cookie_lifetime' => 60*60*24*7, //7 days cookie lifetime
  'session_lifetime' => 60 * 60 * 24, //the lifetime of a session after inactivity
  'auto_logout' => true, //automatic logout after session_lifetime
  'skeletondirectory' => '', //default file  at building time
  'lost_password_link' => 'disabled', //reset password for read only user
  'overwriteprotocol' => '',
  'overwritewebroot' => '/boxhub',
  'overwritehost' => '',
  'appcodechecker' => true,
  'config_is_read_only' => true,
  'log_type' => 'syslog',
);
